code/modules/antagonists/traitor/datum_traitor.dm

```diff
@@ -329,7 +329,12 @@
 
 /datum/outfit/traitor/post_equip(mob/living/carbon/human/H, visualsOnly)
 	var/obj/item/melee/energy/sword/sword = locate() in H.held_items
-	sword.icon_state = "e_sword_on_red"
-	sword.worn_icon_state = "e_sword_on_red"
+	if(sword.flags_1 & INITIALIZED_1)
+		sword.attack_self()
+	else
```

Comment:
```suggestion
	else //Atoms aren't initialized during the screenshots unit test, so we can't call attack_self for it since the sword doesn't have the transforming weapon component yet. The line above just makes the outfit functional in the outfits panel too.
```